backend/corpora/dataset_processing/process.py

```diff
@@ -165,47 +163,51 @@ def check_env():
 
 
 def create_artifact(
-    file_name: str, artifact_type: DatasetArtifactFileType, bucket_prefix: str, dataset_id: str, artifact_bucket: str
+        file_name: str, artifact_type: DatasetArtifactFileType, bucket_prefix: str, dataset_id: str,
+        artifact_bucket: str, processing_status_type: str
 ):
+    update_db(dataset_id, processing_status={processing_status_type: ConversionStatus.UPLOADING})
     logger.info(f"Uploading [{dataset_id}/{file_name}] to S3 bucket: [{artifact_bucket}].")
-    s3_uri = DatasetAsset.upload(file_name, bucket_prefix, artifact_bucket)
-    with db_session_manager() as session:
-        logger.info(f"Updating database with  {artifact_type}.")
-        DatasetAsset.create(
-            session,
-            dataset_id=dataset_id,
-            filename=file_name,
-            filetype=artifact_type,
-            type_enum=DatasetArtifactType.REMIX,
-            user_submitted=True,
-            s3_uri=s3_uri,
-        )
+    try:
+        s3_uri = DatasetAsset.upload(file_name, bucket_prefix, artifact_bucket)
+        with db_session_manager() as session:
+            logger.info(f"Updating database with  {artifact_type}.")
+            DatasetAsset.create(
+                session,
+                dataset_id=dataset_id,
+                filename=file_name,
+                filetype=artifact_type,
+                type_enum=DatasetArtifactType.REMIX,
+                user_submitted=True,
+                s3_uri=s3_uri,
+            )
+        update_db(dataset_id, processing_status={processing_status_type: ConversionStatus.UPLOADED})
+
+    except Exception as e:
```

Comment:
Shall we maybe put the message of the exception in the database as well? It will make debugging even easier.